docs/appnotes/0015-using-tams-in-opentimelineio.md

```diff
@@ -0,0 +1,171 @@
+
+### MissingReference Format
+
+This format uses metadata to reference a Source ID in TAMS.
+In this example the reference is a `MissingReference`: no specific location is given for the media (since a Source has no concrete location).
+A client with access to a suitable TAMS instance could use a [Media Linker](https://opentimelineio.readthedocs.io/en/latest/tutorials/write-a-media-linker.html) plugin to read the `metadata` dictionary and locate the Source, and then select suitable a subset of Flows.
+
+Once linked, an `ExternalReference` with a `target_url` could be constructed and used for each Flow as below (with some suitable mechanism used to set the `active_media_reference_key`), which retains the same `metadata` dictionary.
+In principle the `metadata` [should be preserved](https://github.com/AcademySoftwareFoundation/OpenTimelineIO/wiki/OpenTimelineIO-Application-Integrator's-Guide#preserve-metadata-to-the-best-of-your-abilities) so even if the `target_url` is replaced with a file on disk, it should still be possible to reconstruct the TAMS reference.
+
+The `start_time` and `duration` in the `available_range` are Flow timestamps with nanosecond precision, referring to the timerange over which the Flow is available.
+The `source_range` also has a `start_time` as a nanosecond timestamp within the same Flow.
+
+```json
+"media_references": {
+    "DEFAULT_MEDIA": {
+        "OTIO_SCHEMA": "MissingReference.1",
+        "metadata": {
+            "bbc.github.io/tams": {
+                "source_id": "9bb414a5-862c-494f-86ce-8e2720ecc315",
+                "available_range_offset": "0:0"
+            }
+        },
+        "name": "camera-one.ts",
+        "available_range": {
+            "OTIO_SCHEMA": "TimeRange.1",
+            "duration": {
+                "OTIO_SCHEMA": "RationalTime.1",
+                "rate": 1000000000.0,
+                "value": 372100000000.0
+            },
+            "start_time": {
+                "OTIO_SCHEMA": "RationalTime.1",
+                "rate": 1000000000.0,
+                "value": 1723124086620000000
+            }
+        },
+        "available_image_bounds": null
+    }
+}
+```
+
+Note the presence of an `available_range_offset` that describes how the `available_range` here maps onto the Flow timeline, much as `ts_offset` remaps media essence timing to Flow timing.
+
+### ExternalReference Format
+
+This format refers to a URL in a TAMS instance and points to a specific, concrete Flow.
+It is possible a Clip could have more than one `ExternalReference` object (e.g. for each Flow), with the appropriate default selected: this allows, for example, online and proxy workflows to be presented in a UI.
+
+Notice that the URL has a prefix `tamss://` (for "TAMS Secure" - `tams://` would also work for HTTP).
```

Comment:
I wonder whether it is simpler to just have a `tams://` pseudo-protocol. If secure/non-secure is an option then I don't see why this `tams` locator needs to decide either way - that's for the linker to decide when translating to the actual protocol.

Is it worth stating why you would use `tams://` rather than `https://`, given that they look essentially the same?

Given my comments on 0014, one option would be to separate out the TAMS location (e.g. the base url). This might make it easier for linkers that want or need to use a different location to have the identity part separate from the (assumed / default) location. However, it's just a thought and I'm not advocating changing the design here.